src/libs/actions/Transaction.js

```diff
@@ -124,7 +116,8 @@ function removeWaypoint(transactionID, currentIndex) {
     }
 
     const waypointValues = _.values(existingWaypoints);
-    waypointValues.splice(index, 1);
+    const removed = waypointValues.splice(index, 1);
```

Comment:
Coming from https://github.com/Expensify/App/issues/30176
The route was rerendered if you add an empty stop, we resolved this by returning early if there are no removed waypoints, i.e.
```js
    if (removed.length === 0) {
        return;
    }
   ```